services/src/main/java/org/exoplatform/task/util/TaskUtil.java

```diff
@@ -781,6 +781,10 @@ public static Event buildEvent(Event event, Task task) {
       //TODO: how to process localization for status name when build calendar event
       event.setEventState(encoder.encode(task.getStatus().getName()));
     }
+    else {
+      //usage of default task calendar when no status is set
+      event.setCalendarId(String.valueOf(ProjectUtil.TODO_PROJECT_ID));
```

Comment:
Please add unit tests to check this case.

Reply:
With this change, all task events will have the same calendar id if I understood correctly. Could it be a problem ? Did you test with multiple users ? (please update the PR description to explain the changes).

Reply:
The Tasks without project belongs to the default task calendar. The tasks within projects belong to project's